packages/create/src/cli/change/change-data-source.ts

```diff
@@ -0,0 +1,100 @@
+    cmd.command('data-source [path]')
+        .option('-s, --simulate', 'simulate only do not write or install')
+        .action(async (path, options) => {
+            await changeDataSource(path, { ...options }, !!options.simulate);
+        });
+}
+
+/**
+ * Changes the data source of an adaptation project.
+ *
+ * @param {string} basePath - The path to the adaptation project.
+ * @param {PromptDefaults} defaults - The default values for the prompts.
+ * @param {boolean} simulate - If set to true, then no files will be written to the filesystem.
+ */
+async function changeDataSource(basePath: string, defaults: PromptDefaults, simulate: boolean): Promise<void> {
+    const logger = getLogger();
+    try {
+        if (!basePath) {
+            basePath = process.cwd();
+        }
+        checkEnvironment(basePath);
+
+        const variant = getVariant(basePath);
+        const ui5Conf = await UI5Config.newInstance(readFileSync(join(basePath, 'ui5.yaml'), 'utf-8'));
+        const adp = ui5Conf.findCustomMiddleware<{ adp: AdpPreviewConfig }>('fiori-tools-preview')?.configuration?.adp;
+        if (!adp) {
+            throw new Error('No system configuration found in ui5.yaml');
+        }
+        const manifest = await getManifest(variant.reference, adp, logger);
+        const dataSources = manifest['sap.app'].dataSources;
+        if (!dataSources) {
+            throw new Error('No data sources found in the manifest');
+        }
+        const answers = await promptYUIQuestions(getPromptsForChangeDataSource(dataSources), false);
+
+        const fs = await generateChange<ChangeType.CHANGE_DATA_SOURCE>(basePath, ChangeType.CHANGE_DATA_SOURCE, {
+            variant,
+            dataSources,
+            answers
+        });
+
+        if (!simulate) {
+            await new Promise((resolve) => fs.commit(resolve));
+        } else {
+            await traceChanges(fs);
+        }
+    } catch (error) {
+        logger.error(error.message);
+        if (error.response?.status === 401 && loginAttempts) {
+            loginAttempts--;
+            logger.error(`Authentication failed. Please check your credentials. Login attempts left: ${loginAttempts}`);
+            await changeDataSource(basePath, defaults, simulate);
+            return;
+        }
+        logger.debug(error);
+    }
+}
+
+/**
+ * Get the app descriptor variant.
+ *
+ * @param {string} basePath - The path to the adaptation project.
+ * @returns {DescriptorVariant} The app descriptor variant.
+ */
+function getVariant(basePath: string): DescriptorVariant {
+    return JSON.parse(readFileSync(join(basePath, 'webapp', 'manifest.appdescr_variant'), 'utf-8'));
+}
+
+/**
+ * Check if the project is a CF project.
+ *
+ * @param {string} basePath - The path to the adaptation project.
+ * @throws {Error} If the project is a CF project.
+ */
+function checkEnvironment(basePath: string): void {
+    const configJsonPath = join(basePath, '.adp', 'config.json');
```

Comment:
I would also check if the project actually is an adp and not an app - I think `project-access` has a function to check if it is adp

Reply:
I checked the whole `open-ux-tools` repo but I cannot find such a function.

Reply:
@Klaus-Keller did I make that up? I thought we have a function in the `project-access` to identify ADPs?

Reply:
I think I found the function you mentioned in `project-access`. I tested locally and it seems to be working as expected. Could you please take a look at: 4a3eb7207c181990d27a88c33639ca8d43c98d95

Reply:
Looks good!